src/stdlib/parse_float.rs

```diff
@@ -0,0 +1,142 @@
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "integer",
+                source: r#"parse_float!("-42")"#,
+                result: Ok("-42.0"),
+            },
+            Example {
+                title: "float",
+                source: r#"parse_float!("42.38")"#,
+                result: Ok("42.38"),
+            },
+            Example {
+                title: "scientific notation",
+                source: r#"parse_float!("2.5e3")"#,
+                result: Ok("2500.0"),
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(ParseIntFn { value }.as_expr())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseIntFn {
+    value: Box<dyn Expression>,
+}
+
+impl FunctionExpression for ParseIntFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+
+        parse_float(value)
+    }
+
+    fn type_def(&self, _: &state::TypeState) -> TypeDef {
+        TypeDef::float().fallible()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    test_function![
+        parse_float => ParseFloat;
+
+        integer {
+            args: func_args![value: "-42"],
+            want: Ok(-42.0),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        float {
+            args: func_args![value: "42.38"],
+            want: Ok(42.38),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        scientific_1 {
+            args: func_args![value: "2.5e3"],
+            want: Ok(2500.0),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        scientific_2 {
+            args: func_args![value: "8.543e-2"],
+            want: Ok(0.08543),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        positive_zero {
+            args: func_args![value: "+0"],
+            want: Ok(0.0),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        negative_zero {
+            args: func_args![value: "-0"],
+            want: Ok(-0.0),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        positive_infinity {
+            args: func_args![value: "inf"],
+            want: Ok(f64::INFINITY),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        negative_infinity {
+            args: func_args![value: "-inf"],
+            want: Ok(f64::NEG_INFINITY),
+            tdef: TypeDef::float().fallible(),
+        }
+
+        nan {
+            args: func_args![value: "Nan"],
+            want: Err("NaN number not supported".into()),
+            tdef: TypeDef::float().fallible(),
+        }
+    ];
```

Comment:
Let's add two cases for [MAX](https://doc.rust-lang.org/std/primitive.f64.html#associatedconstant.MAX) and [MIN](https://doc.rust-lang.org/std/primitive.f64.html#associatedconstant.MIN). Since these special values tend to be a common source of bugs.